### PR TITLE
Add quickstart sample for AKS private cluster with custom DNS

### PR DESCRIPTION
**Pull request by feiskyer**
Add a quick-start sample for AKS private cluster with custom DNS.

### CODE REVIEW

File: quickstart/301-aks-private-cluster/dns-zone-link.sh [hunk 1]
```diff
@@ -0,0 +1,35 @@
+#!/bin/bash
+# Query AKS private DNS zone and then link it to custom DNS VNet.
```
Comment (grayzu) [inline quickstart/301-aks-private-cluster/dns-zone-link.sh:2]: Why does this need to be done with a script? Can't you not use a combination of the azurerm_* and the dns_* providers to accomplish this?

**Review by grayzu — COMMENTED** [review]
Mostly looks great. One question regarding the script. It would be great if this could be done without the script and just use native providers.

File: quickstart/301-aks-private-cluster/dns-zone-link.sh (hunk 1 above)
Comment (feiskyer) [reply quickstart/301-aks-private-cluster/dns-zone-link.sh:2]: because the private dns zone name is unknown until we query it here.

**Comment by feiskyer** [comment]
@grayzu could you help to have a look again?

**Comment by dejiaja** [comment]
@feiskyer thanks. this works as expected except when there is an error with the cluster provisioning. When this is run via Azure devops for instance and there is an error with the AKS resource (say the resource already exists), I would want the wait for resource group to timeout and display the aks failure error but the the wait for resource group keeps running till you cancel the release. I'm experimenting with the timeout but havent gotten this to work as desired. Any recommendations?

**Comment by feiskyer** [comment]
@dejiaja what errors have you got when you said "an error with AKS resource"?

**Pull request merged**